anago

```diff
@@ -200,6 +200,46 @@ check_acls () {
 }
 
 ###############################################################################
+# Ensures all registries that will be used during both mock and --nomock
+# runs allow write access so we don't fall over later
+# @param registries - A space separated list of registries
+#
+ensure_registry_acls () {
+  local registries=($1)
+  local emptyfile="/tmp/empty-file.$$"
+  local gs_path
+  local r
+  local retcode=0
+
+  # Make an empty file to test uploading
+  logrun touch $emptyfile
+
+  # If G_AUTH_USER is set, switch to it for verifying connectivity to registries
+  [[ -n $G_AUTH_USER ]] && logrun $GCLOUD config set account $G_AUTH_USER
+
+  # Short of creating a hardcoded map of project-id to registry, translating
+  # _ to - seems to be a simple rule to keep this, well, simple.
+  for r in ${registries[*]//_/-}; do
+    gs_path="gs://artifacts.$r.appspot.com/containers"
```

Comment:
This should work for now.

The proper way to do this would be to perform the token exchange with appropriate scopes:
https://docs.docker.com/registry/spec/auth/token/#requesting-a-token